tests/unit/python/foglamp/services/common/test_microservice.py

```diff
@@ -0,0 +1,164 @@
+
+    def test_constructor_exception(self):
+        class FoglampMicroserviceImp(FoglampMicroservice):
+            def run(self):
+                pass
+            async def change(self):
+                pass
+            async def shutdown(self):
+                pass
+        with patch.object(SilentArgParse, 'silent_arg_parse', side_effect=['corehost', 0, 'sname']):
+            with patch.object(MicroserviceManagementClient, '__init__', return_value=None) as mmc_patch:
+                with patch.object(ReadingsStorageClient, '__init__', return_value=None) as rsc_patch:
+                    with patch.object(StorageClient, '__init__', return_value=None) as sc_patch:
+                        with patch.object(FoglampMicroservice, '_make_microservice_management_app', side_effect=Exception()) as make_patch:
+                            with patch.object(_logger, 'exception') as logger_patch:
+                                with pytest.raises(Exception) as excinfo:
+                                    fm = FoglampMicroserviceImp()
+        logger_patch.assert_called_once_with('Unable to intialize FoglampMicroservice due to exception %s', '')
+
+    @pytest.mark.asyncio
+    async def test_ping(self):
+        class FoglampMicroserviceImp(FoglampMicroservice):
+            def run(self):
+                pass
+            async def change(self):
+                pass
+            async def shutdown(self):
+                pass
+        with patch.object(SilentArgParse, 'silent_arg_parse', side_effect=['corehost', 0, 'sname']):
+            with patch.object(MicroserviceManagementClient, '__init__', return_value=None) as mmc_patch:
+                with patch.object(ReadingsStorageClient, '__init__', return_value=None) as rsc_patch:
+                    with patch.object(StorageClient, '__init__', return_value=None) as sc_patch:
+                        with patch.object(FoglampMicroservice, '_make_microservice_management_app', return_value=None) as make_patch:
+                             with patch.object(FoglampMicroservice, '_run_microservice_management_app', side_effect=None) as run_patch:
+                                 with patch.object(FoglampProcess, 'register_service_with_core', return_value={'id':'bla'}) as reg_patch:
+                                     with patch.object(FoglampMicroservice, '_get_service_registration_payload', return_value=None) as payload_patch:
+                                         with patch.object(web, 'json_response', return_value=None) as response_patch:
+                                             # called once on FoglampProcess init for _start_time, once for ping
+                                             with patch.object(time, 'time', return_value=1) as time_patch:
+                                                 fm = FoglampMicroserviceImp()
+                                                 await fm.ping(None)
+        response_patch.assert_called_once_with({'uptime': 0})
+
```

Comment:
Not sure, if we should stub `change` and `shutdown` endpoint in FoglampMicroservice**Imp** and make calls to them too?!